src/operator/cudnn_ops.cc

```diff
@@ -0,0 +1,765 @@
+  }
+}
+
+std::vector<Descriptor> MakeFallbackPlans(
+    const std::vector<int64_t>& ixs,
+    cudnnHandle_t handle,
+    const Descriptor& op_graph,
+    size_t workspace_limit,
+    size_t* max_workspace,
+    const std::unordered_set<int64_t>& excl_engines,
+    const std::vector<cudnnBackendNumericalNote_t>& req_numeric,
+    const std::vector<cudnnBackendNumericalNote_t>& excl_numeric
+#if CUDNN_VERSION >= 8200
+    ,
+    const std::vector<cudnnBackendBehaviorNote_t>& req_behavior,
+    const std::vector<cudnnBackendBehaviorNote_t>& excl_behavior
+#endif  // CUDNN_VERSION >= 8200
+) {
+  std::vector<Descriptor> plans;
+  if (max_workspace)
+    *max_workspace = 0;
+  for (auto ix : ixs) {
+    if (excl_engines.count(ix))
+      continue;
+    auto engine = Make(CUDNN_BACKEND_ENGINE_DESCRIPTOR,
+                       CUDNN_ATTR_ENGINE_OPERATION_GRAPH,
+                       op_graph,
+                       CUDNN_ATTR_ENGINE_GLOBAL_INDEX,
+                       ix);
+    auto err    = cudnnBackendFinalize(engine.get());
+    if (err == CUDNN_STATUS_NOT_SUPPORTED || err == CUDNN_STATUS_ARCH_MISMATCH)
+      continue;
+    if (err != CUDNN_STATUS_SUCCESS) {
+      LOG(WARNING) << "Unexpected cuDNN status: " << err << ": " << cudnnGetErrorString(err);
+      continue;
+    }
+    auto cfg =
+        MakeFinalized(CUDNN_BACKEND_ENGINECFG_DESCRIPTOR, CUDNN_ATTR_ENGINECFG_ENGINE, engine);
+    auto plan = Make(CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR,
+                     CUDNN_ATTR_EXECUTION_PLAN_HANDLE,
+                     handle,
+                     CUDNN_ATTR_EXECUTION_PLAN_ENGINE_CONFIG,
+                     cfg);
+    err       = cudnnBackendFinalize(plan.get());
+    if (err == CUDNN_STATUS_NOT_SUPPORTED || err == CUDNN_STATUS_ARCH_MISMATCH)
+      continue;
+    if (err != CUDNN_STATUS_SUCCESS) {
+      LOG(WARNING) << "Unexpected cuDNN status: " << err << ": " << cudnnGetErrorString(err);
+      continue;
+    }
+    auto workspace = GetAttr<int64_t>(plan, CUDNN_ATTR_EXECUTION_PLAN_WORKSPACE_SIZE);
+    if (workspace > workspace_limit)
+      continue;
+    auto numerical = GetSomeAttrs<cudnnBackendNumericalNote_t>(
+        CUDNN_NUMERICAL_NOTE_TYPE_COUNT, engine, CUDNN_ATTR_ENGINE_NUMERICAL_NOTE);
+    if (!IsCompatible(numerical, req_numeric, excl_numeric))
+      continue;
+#if CUDNN_VERSION >= 8200
+    auto behavior = GetSomeAttrs<cudnnBackendBehaviorNote_t>(
+        CUDNN_BEHAVIOR_NOTE_TYPE_COUNT, engine, CUDNN_ATTR_ENGINE_BEHAVIOR_NOTE);
+    if (!IsCompatible(behavior, req_behavior, excl_behavior))
+      continue;
+#endif  // CUDNN_VERSION >= 8200
+    plans.push_back(std::move(plan));
+    if (max_workspace)
+      *max_workspace = std::max(*max_workspace, static_cast<size_t>(workspace));
+  }
+  return plans;
+}
+
+cudnnBackendHeurMode_t HeurMode() {
+#if CUDNN_VERSION >= 8100
+  auto minor       = cudnnGetVersion() / 100 % 10;
+  int default_mode = minor < 2 ? CUDNN_HEUR_MODE_INSTANT : CUDNN_HEUR_MODE_B;
```

Comment:
I fixed the code to account for future major version numbers.
The current logic is: MODE_B is the default if both compile time and runtime version is at least 8100, otherwise it's MODE_INSTANT.